lib/query/where.ex

```diff
@@ -185,4 +193,19 @@ defmodule QueryBuilder.Query.Where do
         end
     end
   end
+  defp value_is_field(val) when val in [nil, false, true], do: false
+
+  defp value_is_field(val) when is_atom(val) do
+    val |> Atom.to_string() |> String.ends_with?(@value_is_field_marker)
+  end
+
+  defp value_is_field(_), do: false
+
+  defp referenced_field_in_value(val) when is_atom(val) do
+    str_val = Atom.to_string(val)
+
+    str_val
+    |> binary_part(0, byte_size(str_val) - byte_size(@value_is_field_marker))
+    |> String.to_atom()
```

Comment:
I think we can use `to_existing_atom` here? I assume you know about the danger of `to_atom`.

Reply:
@mathieuprog removed the atom conversion entirely.  That's performed further down the implementation chain anyway